src/liberty/libertyProject.ts

```diff
@@ -338,13 +338,21 @@ export class ProjectProvider implements vscode.TreeDataProvider<LibertyProject>
 				if (project.contextValue !== projectType) {
 					project.setContextValue(projectType);
 				}
+				fse.readFile(buildFilePath, "utf8").then(value => {
```

Comment:
It looks like this async block is causing the labels to not be automatically updated, since `setLabel` is being called after the project is added to the `projectsMap`. Details discussed over Slack.